### ISSUE TITLE
grub4dos legacy: echo bug with $[] in mode 3

### ISSUE THREAD

**Issue by steve6375** [chenall/grub4dos#381]
[text.zip](https://github.com/chenall/grub4dos/files/9692566/text.zip)
seems to be an issue in mode 3 when using echo with  $[]
![image](https://user-images.githubusercontent.com/1469350/193455634-4aa3c20a-0bf6-4fff-bc6b-def170caf16b.png)
the clear command helps but does not cure it.
Seems OK in 800x600 or 1024x768 graphicsmode

I tested grub4efi but seems to hang??

**Comment by yaya2007** [comment]
I used the QEMU virtual machine to test, and both the bios and uefi environments worked well.

**Comment by steve6375** [comment]
QEMU BIOS does not show problem.
Real systems and VirtualBox do show the problem.

**Comment by yaya2007** [comment]
echo $[0xf0]1234567890$[]1234567890

旧版本grub4dos，在非QEMU虚拟机，在控制台模式（80*25）的0—23行执行，没有一点问题。但是在第24行执行就暴露问题了。尤其是字符在80个以上时。
分析原因：
1. 滚屏时，首先把1—24行字符移动到0—23行。
2. 打印80个空格，清除第24行内容。
3. 打印新的内容。
问题就出在第二条。它使用第24行第一个字符的属性打印，这时属性是白色背景！

如果第一个字符是常规属性，就不会有问题了。

**Comment by yaya2007** [comment]
也许有一个更好的办法，就是在第24行时再起作用（跳过0列，在第1列打印）。
[grldr.rar.txt](https://github.com/chenall/grub4dos/files/9949253/grldr.rar.txt)

**Comment by steve6375** [comment]
New version seems OK
P.S.
I dont know why you talk about scrolling?? Here is yesterdays grldr with no \menu.lst
The screen is not scrolling and yet it prints in column 1
![image](https://user-images.githubusercontent.com/1469350/200281196-cd55521b-a77b-437c-97f6-ce22ae4c807b.png)

`P.S. Why is title and text printed twice ?` << IGNORE!

**Comment by yaya2007** [comment]
I'm not sure. Let's test it tomorrow.

The scrolling problem. You used the version test on September 15, and the command line was repeated 30 times.

**Comment by yaya2007** [comment]
You can use the test.txt on the first floor to test it.

**Comment by steve6375** [comment]
P.S. Why is title and text printed twice ? I was using 'commandline' command so I got two messages. Ignore this. not a problem.

**Comment by yaya2007** [comment]
I have no problem testing here. It is definitely not a problem introduced this time. You can test several different types of virtual machines. Or test on a real machine.

**Comment by steve6375** [comment]
> I have no problem testing here. It is definitely not a problem introduced this time. You can test several different types of virtual machines. Or test on a real machine.

Yes. ignore this. I was using menu.lst with 'commandline' as first line and so it was 'commandline' that was causing the extra text. sorry :-(

**Comment by steve6375** [comment]
[TESTECHO.zip](https://github.com/chenall/grub4dos/files/9959771/TESTECHO.zip)
In graphicsmode 3

Run once is OK with no scrolling
Run again with crolling and get printing in column 1
![image](https://user-images.githubusercontent.com/1469350/200531270-2b2ee779-5557-47cc-bda0-996bf18dc1a4.png)

![image](https://user-images.githubusercontent.com/1469350/200531153-7aa22f19-7073-4936-927b-b1022d39bfaf.png)
